### DISCUSSION TITLE
Documentation versus blog

### DISCUSSION THREAD

**Discussion by typxxi** [kotope/aio_energy_management#26] [Ideas]
I am installing the system right now. I had read along the blog articles for 2 years , played with it in the beginning and then had to build my big solar array, battery, inverter farm and wallbox.

Not that I am back I was reading again in the latest blog articles and started with the nordpool integration article which I had been able to fix yesterday cause nordpool had been broken in germany due to wrong VAT %. 

So I read the article and then opened hacs with the link provided and ended up in a new and differen documentation. 

So I came from the article and installed, then did a reboot and meanwhile I got lost cause I had read about the yaml configuration and found it weird that there was entsoe part of the yaml code. So I copied that code into notepad++ to prepare that for integration in my configuration.yaml

And here I got lost cause when I went back for the guide I ended up in the blog article where I found different yaml code. Not really cause that was the pure nordpool yaaml code, but it took me quite a while to find out that the HACS readme page is different to the blog. I simply could not find the yaml code with the 3 sensors till I checked HA for the reboot.

I suggest to separate the sensors in the hacs documentation to make it clear that all the lines from the base integration (nordpool or entso-e) should be removed. 

Finally the documentation is unclear in one point regarding the helper called input_number.my_input_number
Why is this used only for the most expensive hours but not for the cheapest ? 
At least a bit irritating and would be better to use that helper there too.

**Comment by kotope** [comment]
You got a good points here and let me try to explain my point of view :-)

The problem with blogging is that the articles get easily outdated, but the project README.md should be always up to date. While writing the first artilcle of AIO Energy Management I had not implemented Entso-E yet so that's definately missing from the first article. I wrote a second article later about entso-e support, but haven't linked it on the first article.
And the reason I'm linking from my GitHub ReadME.md to the blog is purely because I want to forward people to my blog obviously as that's my main '_business_'.

A one solution for this confusion could be for me to change the readme link to point to AIO Energy Management 'category' on my blog so people could see all the posts related to AIO Energy Management rather than just the first one that is now somewhat old already.

And what comes to the input_number, I tried to make an example to cover all the configuration possibilities and decided just to add the input_number to the expensive hours, there's nothing special on it what so ever. 
I've got some feedback of non-working integration mostly because people have just copy-pasted the example and haven't read or thought of the requirement to actually have the input_number helper defined somewhere. I will remove that from the example on README.md on the next release and leave it visible only on the attribute table. Hopefully people can still  notice it from there :-)

**Comment by typxxi** [comment]
Well,
right now the issues have gone or the insecurity cause 1 sensor was missing till 6 pm and of cause the important one for the cheapest hours.

I like your blog and I came to the whole nordpool topic via google search engine when tibber was broken due to matplot lib missing and no one really cared about complaints - till today cause I still get a boot error caused by tibber HA integration even though the price sensor is working again. But only for easy comparison if Nordpool and my formula match the tibber app and tibber sensor values.

ANd of cause if you offer the easy way with links like these blue button to add an integration and so on people  get an easy way. 

So I suggest you write a  line before people can click that they will being forwarded to the most recent description and guide on github which might be a bit different due to changes since the blog article had been published. 

I was aware of your "aio development history, I had read them all I guess, so from the 2022 beginning over the iterations and then the big leap into the integration "package" or however it is called up to entso-e, but of cause not looking into the tiny code or example details. 

The blog is very usefull and also entertaining, explains lot more and deeper with more aspects. 
It is just a reminder to welcome those which might come from an older blog article nad might have missed something. 

I got everything fixed so far and it looks good to start with except that we had not much sun or rather 5 minutes and then thick clouds till later afternoon when the suns power was weak with 1,5 kW at peak instead of up to 25 kW in summer. 

Regarding the inputnumber some things are easy, others are more complicated and of cause you have lowered the likelihood of risks by using both ways: hours from the yaml file or from the input helper

But at 1 point while working step by step throught the guide it still made me wonder why 1 input helper and then the figure in the yaml code cause it raised questions why.  

And with your comment it has become to me far clearer how complicated the situation is for you coming from a blog and its history and the whole evoultion to a most recent 1 page guide for all on github (which I was not aware of cause I had assumed 2 blog articles would end up in 2 different AIO releases on github where I was wrong and the 3rd part of the entso-e sensors showed cause those would be needed only for entso-e users. 

Thanks for you quick reply.
